libs/ts-rest/nest/src/lib/ts-rest-nest.ts

```diff
@@ -1,33 +1,48 @@
-import { AppRoute, AppRouter, ApiRouteResponse, Without, getRouteResponses } from '@ts-rest/core';
-import { ApiDecoratorShape } from './api.decorator';
+import {
+  AppRoute,
+  AppRouter,
+  ApiRouteResponse,
+  Without,
+  getRouteResponses,
+  ApiResponseForRoute,
+} from '@ts-rest/core';
+import { TypedRequestShape } from './typed-request.decorator';
 
 type AppRouterMethodShape<T extends AppRoute> = (
   ...args: any[]
 ) => Promise<ApiRouteResponse<T['responses']>>;
 
-type AppRouterControllerShape<T extends AppRouter> = {
-  [K in keyof T]: T[K] extends AppRouter
-    ? undefined
-    : T[K] extends AppRoute
-    ? AppRouterMethodShape<T[K]>
-    : never;
-};
+type AppRouterControllerShape<T extends AppRouter> = Without<
+  {
+    [K in keyof T]: T[K] extends AppRoute ? AppRouterMethodShape<T[K]> : never;
+  },
+  never
+>;
 
-type AppRouteShape<T extends AppRouter> = {
-  [K in keyof T]: T[K] extends AppRouter
-    ? AppRouteShape<T[K]>
-    : T[K] extends AppRoute
-    ? ApiDecoratorShape<T[K]>
-    : never;
-};
+type AppRouterRequestShapes<T extends AppRouter> = Without<
+  {
+    [K in keyof T]: T[K] extends AppRoute ? TypedRequestShape<T[K]> : never;
+  },
+  never
+>;
+
+type AppRouterResponseShapes<T extends AppRouter> = Without<
+  {
+    [K in keyof T]: T[K] extends AppRoute ? ApiResponseForRoute<T[K]> : never;
+  },
+  never
+>;
 
-export type NestControllerShapeFromAppRouter<T extends AppRouter> = Without<
+type NestControllerShapeFromAppRouter<T extends AppRouter> = Without<
   AppRouterControllerShape<T>,
   AppRouter
 >;
 
-export type NestAppRouteShape<T extends AppRouter> = AppRouteShape<T>;
+type NestAppRouteShape<T extends AppRouter> = AppRouterRequestShapes<T>;
 
+/**
+ * @deprecated Use `nestControllerContract`, `NestControllerInterface`, `NestRequestShapes`, and `NestResponseShapes` instead. Check the docs for more info.
```

Comment:
Can we link to the nest docs to save users some time?

Reply:
Absolutely!